web/src/engine/osk/src/banner/suggestionBanner.ts

```diff
@@ -704,28 +704,35 @@ export class SuggestionBanner extends Banner {
       minWidth: 0,
     }
 
-    let totalWidth = 0;
-    let displayCount = 0;
-
-    let collapsedOptions: BannerSuggestion[] = [];
-
     for (let i=0; i<SuggestionBanner.SUGGESTION_LIMIT; i++) {
       const d = this.options[i];
 
       if(suggestions.length > i) {
         const suggestion = suggestions[i];
         d.update(suggestion, optionFormat);
-        if(d.collapsedWidth < d.expandedWidth) {
-          collapsedOptions.push(d);
-        }
-
-        totalWidth += d.collapsedWidth;
-        displayCount++;
       } else {
         d.update(null, optionFormat);
       }
     }
 
+    this.refreshLayout();
+  }
+
+  readonly refreshLayout = () => {
+    let collapsedOptions: BannerSuggestion[] = [];
+    let totalWidth = 0;
+
+    let displayCount = Math.min(this.currentSuggestions.length, 8);
+    for(let i=0; i < displayCount; i++) {
+      const opt = this.options[i];
```

Comment:
is it possible to do this?

```suggestion
    for(const opt of this.options) {
      const opt = this.options[i];
```

Reply:
`this.options.length` (locked by literal constant to `8`) does not necessarily equal `displayCount` (the number of available and displayable suggestions; 1 if none are available).

That tripped me up a bit when implementing the change.

Reply:
https://github.com/keymanapp/keyman/blob/c2474bd2e525af7352278dc91415b73d47f6a855/web/src/engine/osk/src/banner/suggestionBanner.ts#L381

https://github.com/keymanapp/keyman/blob/c2474bd2e525af7352278dc91415b73d47f6a855/web/src/engine/osk/src/banner/suggestionBanner.ts#L426-L436

We simply hide ones that don't have a useful suggestion to display.

Reply:
Ouch. That's definitely a trap for young future maintainers. Can you add a source comment accordingly?